src/pages/storage/StoragePoolSelector.tsx

```diff
@@ -56,30 +63,47 @@ const StoragePoolSelector: FC<Props> = ({
   }
 
   const getStoragePoolOptions = () => {
-    const options = poolsToUse.map((pool) => {
-      return {
-        label: pool.name,
-        value: pool.name,
-        disabled: false,
-      };
-    });
-    options.unshift({
-      label:
-        poolsToUse.length === 0 ? "No storage pool available" : "Select option",
-      value: "",
-      disabled: true,
-    });
+    const options: CustomSelectOption[] = [];
+    if (poolsToUse) {
+      poolsToUse.forEach((pool) => {
+        options.push({
+          label: <StoragePoolOptionLabel pool={pool} hasMeterBar={false} />,
+          value: pool.name,
+          disabled: false,
+          text: pool.name,
+        });
+      });
+      if (poolsToUse.length === 0) {
+        options.unshift({
+          label: "No storage pool available",
+          value: "",
+          disabled: true,
+        });
+      }
+    }
+
     return options;
   };
 
+  const handleStoragePoolChange = (value: string) => {
+    setValue(value);
+  };
+
   return (
-    <Select
+    <CustomSelect
       id="storage-pool-selector"
       name="pool"
+      aria-label="Storage Pool"
+      searchable="always"
```

Comment:
```suggestion
```
Let's go with the default of `auto` here, maybe? I think we can hide the search if there are less than 5 storage pools available.